torch/_inductor/codegen/triton_utils.py

```diff
@@ -33,6 +37,13 @@ def signature_of(arg: KernelArgType, *, size_dtype: str) -> str:
                 return "fp32"
             else:
                 return new_tye
+        elif V.graph.in_zero_dim_cpu_tensor_list(arg.buffer):
+            if arg.dtype.is_floating_point:
+                return "fp64"
+            elif arg.dtype.is_signed:
+                return "int64"
+            else:
+                return "uint64"
```

Comment:
I think we actually want the existing dtype behavior. The problem is that these fp64 tensors when used as fp64 inside the triton kernel cause triton to upcast the dtypes, which is not what we want. We want the unspec kernel to follow the dtype of the kernel

Reply:
I found inductor just cast the this scalar tensor to the dtype same with other tensors. 
```
    tmp1 = in_ptr1
    tmp2 = tmp1.to(tl.float32)
```
let me remove this part.

Reply:
hmm, true, maybe it was the masked_where previously that led to issues. In any case we should be consistent with unspec_arg and fix both in the future, not diverge behaviors